cpp/src/request.cpp

```diff
@@ -162,20 +162,14 @@ void Request::setStatus(ucs_status_t status)
   if (_endpoint != nullptr) _endpoint->removeInflightRequest(this);
   _worker->removeInflightRequest(this);
 
-  if (_status == UCS_INPROGRESS) {
-    // If the status is not `UCS_INPROGRESS`, the derived class has already set the
-    // status, a truncated message for example.
-    _status.store(status);
-  }
-
-  ucs_status_t s = _status;
-
   ucxx_trace_req_f(_ownerString.c_str(),
                    _request,
                    _operationName.c_str(),
                    "callback called with status %d (%s)",
-                   s,
-                   ucs_status_string(s));
+                   status,
+                   ucs_status_string(status));
+
+  _status.store(status);
```

Comment:
OK so now we just always store the status (even if it is not `INPROGRESS`). Because the derived class might not set the status any more? Or, I guess, it is safer this way.

Reply:
It is safer this way, I wrote the reason in https://github.com/rapidsai/ucxx/pull/58/commits/874b4c377c8d5d00b69800e374623d171141593b but it's indeed not clear without it. Also, TBH, I didn't account for the case where this isn't `UCS_INPROGRESS` as that is not expected to happen, but I've logged an error now in https://github.com/rapidsai/ucxx/pull/58/commits/6f99cbbd436b558ee1c3cc3ec7d8d972e9d9a935 to catch this case should it happen.